scripts/o3de/o3de/export_project.py

```diff
@@ -15,132 +15,198 @@
-        if not os.path.isfile(script_path):
-            logger.error(f"The export script '{script_path}' does not exist!")
-            return False
-
-        if script_path.suffix != '.py':
-            logger.error("A Python script with .py extension must be supplied for --export-script parameter!")
-            return False
-        return True
-    
-    def validate_project_path(project_path):
-        if not os.path.isdir(project_path):
-            logger.error(f"Project path is not a directory!")
-            return False
+    target_folder_path = os.path.split(target_script_path)[0]    
+    if target_folder_path not in sys.path:
+        sys.path.insert(0, target_folder_path)
+
+
+    spec = importlib.util.spec_from_file_location(script_name, target_script_path)
+    script_module = importlib.util.module_from_spec(spec)
+    sys.modules[script_name] = script_module
+
+    o3de_export = O3DEScriptExportParameters()
+    for key,value in kwargs.items():
+        setattr(o3de_export, key, value)
 
-        if not os.path.isfile(os.path.join(project_path, 'project.json')):
-            logger.error("Project path is invalid: does not contain a project.json file!")
-            return False
-        return True
+    if override_logger:
+        parent_logger = o3de_export.logger
+        mark_writeable_params(LOGGER_PARAM)
+        o3de_export.logger = parent_logger.getChild(script_name.replace(".py", ""))
+        mark_readonly_params(LOGGER_PARAM)
 
+    try:
+        spec.loader.exec_module(script_module)
+    except RuntimeError as re:
+        logger.error(f"Failed to run script '{target_script_path}': \nException: "+ str(re))
+        return 1
+    finally:
+        if override_logger:
+            mark_writeable_params(LOGGER_PARAM)
+            o3de_export.logger = parent_logger
+            mark_readonly_params(LOGGER_PARAM)
+
+    return 0
+
+#Export Script entry  point
+def _run_export_script(args: argparse, passthru_args: list) -> int:
+    def get_project_path(args):
+        project_path = args.project_path
+        if not project_path:
+            project_path = utils.find_ancestor_dir_containing_file(pathlib.PurePath('project.json'), export_script_path)
+            if not project_path:
+                logger.error("Unable to find project folder associated with export script. Please specify using --project-path, or select an export script inside a project folder.")
+                return None   
+        
+        if not valid_o3de_project_path(project_path):
+            logger.error(f"Specified Project Path '{project_path}' is not valid.")
+            return None
+        return project_path
+        
+   
+    logger.setLevel(args.log_level)
 
     export_script_path = args.export_script
-    if not validate_export_script(export_script_path):
-        print("Failed to process export script.")
+    if not valid_export_script(export_script_path):
+        logger.error("Failed to process export script.")
```

Comment:
based on the function name, shouldn't the error message read
```suggestion
        logger.error(f"Invalid export script path '{export_script_path}'.")
```